logstash-core/src/main/java/org/logstash/execution/WorkerLoop.java

```diff
@@ -109,17 +111,12 @@ public void run() {
         }
     }
 
-    @SuppressWarnings("unchecked")
-    private void compute(final QueueBatch batch, final boolean flush, final boolean shutdown) {
+    @SuppressWarnings({"unchecked", "rawtypes"})
+    private void compute(final RubyArray batch, final boolean flush, final boolean shutdown) {
```

Comment:
We can probably keep this interface as a `QueueBatch`, and push the `to_a` down into the implementation of `execution#orderedCompute`, which will allow us to avoid the arrayification cost in unordered execution mode.

Reply:
Good idea, let me look at that.

Reply:
actually, ultimately the `compute` method take a `RubyArray` as parameter so conversion from a `QueueBatch` to a `RubyArray` must always happen. Because of that we can't really optimize the unordered case but the ordered could be optimized for the drip loop if we could expose a method that returns a `Collection` instead of using `to_a()` to iterate through all events.

Reply:
I pushed changes for this.  We should definitely followup in another PR to simplify the `AckedReadBatch` impl, I added a comment to this effect.
LMKWYT.